File: examples/python/get_timeline_element_by_name.py
#!/usr/bin/env python

from gi.repository import Gst, GES
from create_timeline_with_n_layers import create_timeline_with_n_layers
from add_one_video_track import add_one_video_track
from play_timeline import play_timeline

if __name__=="__main__":
    Gst.init([])
    GES.init()

    timeline = create_timeline_with_n_layers(1)
    add_one_video_track(timeline)
    layer = timeline.get_layer(0)
    clip = GES.TestClip.new()
    clip.props.duration = 1 * Gst.SECOND
    clip.props.name = "my-awesome-test-clip"
    layer.add_clip(clip)

    got_by_name_clip = timeline.get_element("my-awesome-test-clip")
    got_by_name_clip.props.duration = 5 * Gst.SECOND

    # As usual, the timeline commits itself automatically when going from READY
    # to PAUSED, playback will thus last 5 seconds
    play_timeline(timeline)

File: examples/c/get_timeline_element_by_name.c
#include "examples.h"

int
main (int args, char **argv)
{
  GESTimeline *timeline;
  GESClip *clip, *got_by_name_clip;
  GESLayer *layer;

  gst_init (NULL, NULL);
  ges_init ();

  timeline = create_timeline_with_n_layers (1);
  add_one_video_track (timeline);

  layer = ges_timeline_get_layer (timeline, 0);
  clip = GES_CLIP (ges_test_clip_new ());
  g_object_set (clip, "duration", 1 * GST_SECOND, "name",
      "my-awesome-test-clip", NULL);
  ges_layer_add_clip (layer, clip);
  gst_object_unref (layer);

  got_by_name_clip =
      GES_CLIP (ges_timeline_get_element (timeline, "my-awesome-test-clip"));
  g_object_set (got_by_name_clip, "duration", 5 * GST_SECOND, NULL);

  /* As usual, the timeline commits itself automatically when going from READY
   * to PAUSED, playback will thus last 5 seconds */
  play_timeline (timeline);
  return 0;
}
